### ISSUE TITLE
Godot freezes the whole system when moving windows in a multi monitor setup

### ISSUE THREAD

**Issue by biswas08433** [godotengine/godot#84675]
### Godot version

v4.2.beta5.official [4c96e9676]

### System information

Godot v4.2.beta5 - Pop!_OS 22.04 LTS - X11 - Vulkan (Mobile) - dedicated NVIDIA GeForce GTX 1650 Ti (nvidia; 535.113.01) - AMD Ryzen 5 4600H with Radeon Graphics (12 Threads)

### Issue description

I recently installed a 2nd monitor in my setup. After that godot started freezing my whole system when I start to resize or move the windows. Be it the the primary editor window or the secondary shader editor window. No other application is misbehaving in this way. The freezes only start when I launch godot or move the windows to other display or in the same display

### Steps to reproduce

- Open Godot editor in a multi-monitor setup
- Move or resize the window.

### Minimal reproduction project

[min.zip](https://github.com/godotengine/godot/files/13310914/min.zip)

**Comment by biswas08433** [comment]
It is still present in version 4.2 RC 1

**Comment by Calinou** [comment]
Can you reproduce this after enabling **Single Window Mode** in the Editor Settings?

**Comment by biswas08433** [comment]
Yes it is still happening. I think it has something to do with multiple monitor setup.

**Comment by Calinou** [comment]
Can you reproduce this if you force the editor to start on integrated graphics? You should be able to achieve this using the `--gpu-index N` [command line argument](https://docs.godotengine.org/en/stable/tutorials/editor/command_line_tutorial.html) (try `0`, then `1`).

**Comment by biswas08433** [comment]
No, It seems that I cannot reproduce the issue when using integrated graphics.

**Comment by Calinou** [comment]
In this case, this is likely a NVIDIA graphics driver issue. Support for hybrid graphics on Linux is often sloppy when multi-monitor is involved, especially with NVIDIA :slightly_frowning_face:

**Comment by SamDevelopsCode** [comment]
I am using Linux Mint with x11. I have reproduced this issue on my end to be the power profile for Nvidia. I am using a laptop with Optimus.

When using the On-Demand mode (switches between iGpu and dGpu automatically), Godot will freeze (sometimes freezing my second monitor) when dragging or doing things in the engine.

When set to performance mode (nvidia dGpu only), this never happens. 

I can reproduce reliably.
